packages/test/test-end-to-end-tests/src/test/detachedContainerTests.spec.ts

```diff
@@ -898,4 +898,51 @@ describeNoCompat("Detached Container", (getTestObjectProvider) => {
 			"Stored handle should match!!",
 		);
 	});
+
+	it("Compressors sync after detached container attaches and sends an op", async () => {
+		provider.reset();
+		const testConfig = {
+			...testContainerConfig,
+			runtimeOptions: {
+				...testContainerConfig.runtimeOptions,
+				enableRuntimeIdCompressor: true,
+			},
+		};
+		loader = provider.makeTestLoader(testConfig) as Loader;
+		const container = await loader.createDetachedContainer(provider.defaultCodeDetails);
+
+		// Get the root dataStore from the detached container.
+		const response = await container.request({ url: "/" });
+		const dataStore = response.value as ITestFluidObject;
```

Comment:
nit: You can combine these 2 lines to `const dataStore = requestFluidObject<ITestFluidObject>(container, "/");`